amundsen_application/static/js/components/common/TagsList/index.tsx

```diff
@@ -68,28 +41,36 @@ export class TagsList extends React.Component<TagsListProps, TagsListState> {
   }
 
   render() {
-    let innerContent;
     if (this.props.isLoading) {
-      innerContent = <LoadingSpinner/>;
-    } else {
-      innerContent = (
-        <div id="browse-body" className="browse-body">
-          {this.generateTagInfo(this.state.curatedTags)}
-          {
-            this.state.curatedTags.length > 0 && this.state.otherTags.length > 0 &&
-              <hr />
-          }
-          {this.generateTagInfo(this.state.otherTags)}
+      return <LoadingSpinner/>;
+    }
+    if (AppConfig.browse.curatedTags.length === 0) {
+      return (
+        <div id="tags-list" className="tags-list">
+          { this.generateTagInfo(this.props.allTags) }
         </div>
       );
     }
-    return (innerContent);
+    return (
+      <div id="tags-list" className="tags-list">
+        { this.generateTagInfo(this.props.curatedTags) }
+        {
+          AppConfig.browse.showAllTags && this.props.curatedTags.length > 0 && this.props.otherTags.length > 0 &&
+            <>
+              <hr />
+              { this.generateTagInfo(this.props.otherTags) }
+            </>
+        }
+      </div>
+    );
   }
 }
 
 export const mapStateToProps = (state: GlobalState) => {
   return {
     allTags: state.allTags.allTags,
+    curatedTags: state.allTags.curatedTags,
```

Comment:
I'm not sure there's a need to have all three of these lists passed into the component. At it's core the `TagsList` is a set of primary tags (currently `curatedTags`) and an optional list of secondary tags (currently `otherTags` if the `showAllTags` configuration is true). It seems like while we're here we can just pass a list of `primaryTags` and `secondaryTags` and `TagsList` will render them if they exist. 

The conditional logic of which piece of state gets tied to `primaryTags` and `secondaryTags` could occur here, and would still be based on filtering the one true piece of state -- `allTags`.

Reply:
Creation of `curatedTags` and `otherTags` is moved to the mapStateToProps and removed from reducer/saga